summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -107,13 +132,13 @@ abstract class AsyncBase[I, O, S, D, RC](maxWaitingFutures: MaxWaitingFutures, m
     if (pendingFuturesCount > maxWaitingFuturesCount) {
       // Too many futures waiting, let's clear.
       outstandingFutures.dequeueAll(_.isDefined)
-      val toForce = outstandingFutures.trimTo(maxWaitingFuturesCount).toIndexedSeq
-      if (toForce.nonEmpty) {
+      val toClear = outstandingFutures.size - maxWaitingFuturesCount
+      if (toClear > 0) {
         try {
-          Await.ready(Future.collect(toForce), maxWaitingTime.get)
+          Await.ready(AsyncBase.waitN(outstandingFutures, toClear), maxWaitingTime.get)
         } catch {
           case te: TimeoutException =>
-            logger.error("forceExtra failed on %d Futures".format(toForce.size), te)
+            logger.error("forceExtra failed on %d Futures".format(toClear), te)
```

Comment:
can we use scala interpolation strings: `s"forceExtra failed on $toClear Futures"`